posts/navarroaxel/08.using-event-target-interface.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Using the EventTarget interface
+published: true
+description: Implementing an isomorphic CustomEvent in Node.js v18 and the browser.
+tags: 'javascript, node, webdev, eventdriven'
+cover_image: ./assets/event-target-interface-cover.jpg
+id: 1167436
+---
+
+We have learned how to use the [`CustomEvent`](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent) interface in a [previous post](https://dev.to/cloudx/the-customevent-interface-2bfo).
+
+Now let's build a `fetch` wrapper with a progress event using the CustomEvent interface, which is compatible with Node.js and the browser.
```

Comment:
It might be interesting to actually say WHY you'd want to build a wrapper. What problem are you trying to solve?

Reply:
```suggestion
How can we create a progress indicator using the same JavaScript code for both browser and terminal (using Node.js)? For this we can build a `fetch` wrapper with a progress event using the CustomEvent interface, which is compatible with both environments.
```